packages/ui/src/Tooltip.tsx

```diff
@@ -1,37 +1,141 @@
+
+  /*
+   * We're using useState instead of useRef because react-popper package has some issues with useRef:
+   * https://github.com/popperjs/react-popper/issues/241#issuecomment-591411605
+   */
+  const [referenceElement, setReferenceElement] = useState<HTMLElement | null>(null)
+  const [popperElement, setPopperElement] = useState<HTMLSpanElement | null>(null)
+  const [arrowElement, setArrowElement] = useState<HTMLSpanElement | null>(null)
+
+  const { styles: popperStyles, attributes: popperAttributes } = usePopper(referenceElement, popperElement, {
+    placement,
+    modifiers: [
+      {
+        name: 'arrow',
+        options: {
+          element: arrowElement,
+        },
+      },
+      {
+        name: 'offset',
+        options: {
+          offset: [0, offset],
+        },
+      },
+    ],
+  })
+
+  const onMouseEnter = useCallback(() => {
+    if (hideTimeout !== null) {
+      clearTimeout(hideTimeout)
+    }
+
+    setShown(true)
+  }, [hideTimeout])
+
+  const onMouseLeave = useCallback(() => {
+    setHideTimeout(setTimeout(() => setShown(false), hideTimeoutDuration))
+  }, [hideTimeoutDuration])
+
+  useEffect(() => {
+    referenceElement?.addEventListener('mouseenter', onMouseEnter)
+    popperElement?.addEventListener('mouseenter', onMouseEnter)
+
+    return () => {
+      referenceElement?.removeEventListener('mouseenter', onMouseEnter)
+      popperElement?.removeEventListener('mouseenter', onMouseEnter)
+    }
+  }, [referenceElement, popperElement, onMouseEnter])
+
+  useEffect(() => {
+    referenceElement?.addEventListener('mouseleave', onMouseLeave)
+    popperElement?.addEventListener('mouseleave', onMouseLeave)
+
+    return () => {
+      referenceElement?.removeEventListener('mouseleave', onMouseLeave)
+      popperElement?.removeEventListener('mouseleave', onMouseLeave)
+    }
+  }, [referenceElement, popperElement, onMouseLeave])
+
+  // Makes sure "visible" prop can override local "isShown" state
+  const isTooltipVisible = visibilityOverride ?? isShown
+
+  return (
+    <>
+      {children(setReferenceElement)}
+
+      {content !== undefined && (
+        <>
+          <span id={id} className={styles.tooltipSr} role="tooltip" data-test="tooltip-sr">
+            {content}
+          </span>
+
+          <span
+            ref={setPopperElement}
+            className={cn(styles.overlay, {
+              [styles.hidden]: !isTooltipVisible,
+            })}
+            style={popperStyles.popper}
+            data-test="tooltip-overlay"
+            aria-hidden="true"
+            {...popperAttributes.popper}
+            {...props}
+          >
+            {content}
+
+            <span
+              ref={setArrowElement}
+              className={styles.arrow}
+              style={popperStyles.arrow}
+              data-test="tooltip-arrow"
+              {...popperAttributes.arrow}
+            />
+          </span>
```

Comment:
nit: I like the idea of an element only for screen readers. I think it's unlikely that user somehow stumbles upon both `sr-only` and `non-sr` elements at the same time but shouldn't we hide the `sr-only` element when the tooltip is actually visible? Probably it does not really matter if it's there or not.

Reply:
I think we should never remove `sr-only` element, it's the only description SR users have. Also, we need to keep it because the reference element points to it via `aria-describedby`.